03_letters/patrickmarabeas/main_test.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := fmt.Sprintf("%s\n", "a:2\nb:1")
+	got := buf.String()
+
+	t.Run("Main function", func(t *testing.T) {
+		if expected != got {
+			t.Errorf("\nExpected: %s\nGot:      %s", expected, got)
+		}
+	})
+}
+
+func TestLetters(t *testing.T) {
+	var cases = map[string]struct {
+		input    string
+		expected map[rune]int
+	}{
+		"Letters": {
```

Comment:
you may consider testing more complex combinations such as uppercases and longer strings, as well as empty strings

Reply:
Some additional tests added.